rocAL/rocAL/source/augmentations/geometry_augmentations/node_crop.cpp

```diff
@@ -27,42 +27,43 @@ THE SOFTWARE.
 #include "exception.h"
 
 CropNode::CropNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs) :
-        Node(inputs, outputs),
-        _dest_width(_outputs[0]->info().max_shape()[0]),
-        _dest_height(_outputs[0]->info().max_shape()[1])
-{
+        Node(inputs, outputs) {
     _crop_param = std::make_shared<RocalCropParam>(_batch_size);
 }
 
-void CropNode::create_node()
-{
+void CropNode::create_node() {
     if(_node)
         return;
 
-    if(_dest_width == 0 || _dest_height == 0)
-        THROW("Uninitialized destination dimension")
-
     _crop_param->create_array(_graph);
+    create_crop_tensor(_crop_tensor, &_crop_coordinates);
 
-    // _node = vxExtrppNode_CropPD(_graph->get(), _inputs[0]->handle(), _src_roi_width, _src_roi_height, _outputs[0]->handle(), _crop_param->cropw_arr,
-                                // _crop_param->croph_arr, _crop_param->x1_arr, _crop_param->y1_arr, _batch_size);
-
+    _node = vxRppCrop(_graph->get(), _inputs[0]->handle(), _crop_tensor, _outputs[0]->handle(),
+                      _input_layout, _output_layout, _roi_type);
     vx_status status;
     if((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
-        THROW("Error adding the crop resize node (vxExtrppNode_ResizeCropbatchPD    ) failed: "+TOSTR(status))
+        THROW("Error adding the Crop node (vxRppCrop) failed: "+TOSTR(status))
 }
 
-void CropNode::update_node()
-{
+void CropNode::update_node() {
     _crop_param->set_image_dimensions(_inputs[0]->info().get_roi());
     _crop_param->update_array();
     std::vector<uint32_t> crop_h_dims, crop_w_dims;
     _crop_param->get_crop_dimensions(crop_w_dims, crop_h_dims);
     _outputs[0]->update_tensor_roi(crop_w_dims, crop_h_dims);
+    // Obtain the crop coordinates and update the roi
+    auto x1 = _crop_param->get_x1_arr_val();
+    auto y1 = _crop_param->get_y1_arr_val();
+    RocalROI *crop_dims = static_cast<RocalROI *>_crop_coordinates;
+    for(unsigned i = 0; i < _batch_size; i++) {
+        crop_dims[i].x1 = x1[i];
+        crop_dims[i].y1 = y1[i];
+        crop_dims[i].x2 = crop_w_dims[i];
+        crop_dims[i].y2 = crop_h_dims[i];
```

Comment:
we can move everything to singe line like
crop_dims[i] ={x1[i], y1[i], crop_w_dims[i], crop_h_dims[i]};